generators/ruby-v2/ast/src/ast/core/TypedAstNode.ts

```diff
@@ -0,0 +1,28 @@
+import { AbstractWriter } from "@fern-api/browser-compatible-base-generator";
+import { AbstractFormatter } from "@fern-api/browser-compatible-base-generator";
+
+import { BaseRubyCustomConfigSchema } from "../../custom-config/BaseRubyCustomConfigSchema";
+import { AstNode } from "./AstNode";
+import { RubyFile } from "./RubyFile";
+
+export abstract class TypedAstNode extends AstNode {
```

Comment:
We chatted offline -- but its probably cleaner to rename this to `RubyAstNode` and make `writeTypeDefinition` a noop by default. Anything that has rbs implications can then override the `writeTypeDefinition` function